Lib/test/test_io.py

```diff
@@ -3919,6 +3919,22 @@ def test_issue35928(self):
         f.write(res)
         self.assertEqual(res + f.readline(), 'foo\nbar\n')
 
+    @unittest.skipUnless(hasattr(os, "pipe"), "requires os.pipe()")
+    def test_read_non_blocking(self):
+        import os
+        r, w = os.pipe()
+        try:
+            os.set_blocking(r, False)
+            with open(r, 'r') as textfile:
```

Comment:
TextIOWrapperTest that this is in (https://github.com/python/cpython/blob/main/Lib/test/test_io.py#L2720) is subclassed so it can be run under the two main I/O implementations in python ([`_io`](https://github.com/python/cpython/blob/main/Lib/test/test_io.py#L3942), [`_pyio`](https://github.com/python/cpython/blob/main/Lib/test/test_io.py#L4042)). Here, need to use `self.io.open` to get the implementation-specific `open()` being tested.

For clarity I think it would also be good to do `open(r, 'rt')` rather than relying on "text is the default" since the test is specifically trying to test TextIO behavior

Reply:
Thank you for the insightful suggestion! I've made the recommended changes to ensure that the test runs under both I/O implementations and to clarify the mode being used. The updated code now reads:

```python
with self.io.open(r, 'rt') as textfile:
    r = None
    # Nothing has been written so a non-blocking read raises a BlockingIOError exception.
    with self.assertRaises(BlockingIOError):
```

This should improve clarity and correctness in the context of testing `TextIOWrapper`.